modules/gapi/include/opencv2/gapi/util/variant.hpp

```diff
@@ -428,7 +506,72 @@ namespace util
     {
         return !(lhs == rhs);
     }
-} // namespace cv
+
+namespace detail
+{
+    template<std::size_t CurIndex, std::size_t ElemCount,
+             typename Visitor, typename Variant, typename... VisitorArgs>
+    typename Visitor::result_type apply_visitor_impl(Visitor&& visitor,
+                                                     Variant& v,
+                                                     std::true_type processed,
+                                                     VisitorArgs&& ...args)
+    {
+        suppress_unused_warning(visitor);
+        std::array<bool, sizeof...(VisitorArgs)> dummy{(suppress_unused_warning(args),true)...};
+        suppress_unused_warning(dummy);
+        suppress_unused_warning(v);
+        suppress_unused_warning(processed);
+        return {};
+    }
```

Comment:
What's the purpose of this one?

Reply:
Oh I seem to get it - it just does nothing if a particular variant's type has been visited already?
If it is, then why should we have such an overload? We may possibly terminate the traversal on the match (haven't looked into the below code yet).

Reply:
May be this implementation is not pretty accurate - we can terminate recursion in another way... But in this PR I mostly focused on usability & applicability of visit instead of it's inner implementation

I hope we can use std variant soon or later :) and  i doesn't concentrate on pretty templates implementation here...
But maybe i wrong and turning on c++14 & c++17 is very very far enough